draft-kasselman-oauth_spiffe.md

```diff
@@ -191,6 +191,18 @@ In OAuth flows that rely on redirection, the initial interaction with the author
 * (F)  The client requests an access token from the authorization server's token endpoint by including the authorization code received in the previous step.  When making the request, the client authenticates with the authorization server. It authenticates itself using either a JWT-SVID or X.509-SVID as defined in {{SPIFFE-OAUTH-CLIENT-AUTH}}. The client includes the redirection URI used to obtain the authorization code for verification.
 * (G)  The authorization server authenticates the client, validates the authorization code, and ensures that the redirection URI received matches the URI used to redirect the client in step (E). If valid, the authorization server responds back with an access token and, optionally, a refresh token.
 
+## Client Registration Error Response
+When the registration attempt has failed or been rejected, the authorization server MUST return an error response that conforms to the expected error response for the given OAuth 2.0 endpoint. Additional error information may or may not be included in the error response.
+
+### Authentication Failed
+When a SPIFFE credential is missing, invalid, or for any reason rejected, the authorization server returns an error response with an HTTP 401 status code. Care must be taken to return enough useful information, but not so much as to potentially aid malicious actors.
```

Comment:
>Care must be taken to return enough useful information, but not so much as to potentially aid malicious actors.

I would suggest removing this and instead be explicit about error codes instead or error conditions that are returned (expired, signature not verified, etc). Leaving it vague like this will cause  interop issues. It may be appropriate to add a "security considerations" section and discuss it more there. In the near term this can be left as a TODO.